testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxrs/cfg/resources/ErrorResource.java

```diff
@@ -33,4 +46,35 @@ public Response html(final String html) {
         return Response.status(Response.Status.BAD_REQUEST).entity(html).build();
     }
 
+    /**
+     * Always throws an {@code 401} with a {@code user-id} header entry and a JSON error message indicated by an
+     * {@code error} key.
+     *
+     * @return the error response
+     */
+    @GET
+    @Path("/auth")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response failedAuth() {
+        final String userId = "anonymous";
+        final JsonObjectBuilder builder = Json.createObjectBuilder();
+        throw new WebApplicationException(
```

Comment:
@jamezp isn't `WebApplicationException` it the kind of exception that the client is expected to use to wrap the original exception? I mean, the original exception can be any exception and still, the client has to wrap it in a `WebApplicationException` ... can we generalize the test and throw an exception other than `WebApplicationException`?

Reply:
I see... but IIUC technically the header isn't necessary to test the `resteasy.original.webapplicationexception.behavior` attribute... why are we adding it here?

Reply:
Then my suggestion would be to have two distinct endpoints: one for checking the header (using `WebApplicationException`) and one for checking response sanitizing (not using `WebApplicationException`) ... I suppose it would be easier to spot what breaks (in case it breaks) if we keep them separated ... for sure it would for me when checking the testing pipeline